doc/source/qip-vqa.rst

```diff
@@ -0,0 +1,124 @@
+.. _qip_vqa:
+
+******************************
+Variational Quantum Algorithms
+******************************
+
+Implemented by `Ben Braham <https://benbraham.com>`_ as part of a `Unitary Fund microgrant <https://unitary.fund/grants.html>`_.
+
+Overview
+========
+
+Variational Quantum Algorithms (VQAs) are a hybrid quantum-classical optimization algorithm in which an objective function (usually encoded by a parameterized quantum circuit) is evaluated by quantum computation, and the parameters of this function are updated using classical optimization methods. Such algorithms have been proposed for use in NISQ-era quantum computers as they typically scale well with the number of available qubits, and can function without high fault-tolerance.
+
+In QuTiP, VQAs are represented by a parameterized quantum circuit, and include methods for defining a cost function for the circuit, and finding parameters that minimize this cost.
+
+
+Constructing a VQA circuit
+==========================
+
+The :class:`.VQA` class allows for the construction of a parameterized circuit from :class:`.VQABlock` instances, which act as the gates of the circuit. In the most basic instance, a :class:`.VQA` should have:
+
+====================  =================================================
+Property                           Description
+====================  =================================================
+``n_qubits``          Positive integer number of qubits for the circuit 
+``n_layers``          Positive integer number of repetitions of the 
+                      layered elements of the circuit
+``cost_method``       String referring to the method used to
+                      evaluate the circuit's cost.
+
+                      Either "OBSERVABLE", "BITSTRING", or "STATE".
+====================  =================================================
+
+For example:
+
+.. code-block::
+
+    from qutip_qip.vqa import VQA
+
+    VQA_circuit = VQA(
+                n_qubits=1,
+                n_layers=1,
+                cost_method="OBSERVABLE",
+            )
+
+After constructing this instance, we are ready begin adding elements to our parameterized circuit. Circuit elements in this module are represented by :class:`.VQABlock` instances. Fundamentally, the role of this class is to generate an operator for the circuit. To do this, it keeps track of its free parameters, and gives information to the :class:`.VQA` instance on how to update them. The operator itself can be generated by a user-defined function call, a parameterized Hamiltonian to exponentiate, a pre-computed unitary operator, or a string referring to a gate that has already been defined (either by the user already, or a gate native to QuTiP).
+
+In the absence of specification, a VQA block will assume any :class:`~.Qobj` it is given to be a Hamiltonian, :math:`H`, and will generate a unitary with free parameter, :math:`\theta`, as :math:`U(\theta) = e^{-i \theta H}`. For example, 
```

Comment:
If I understand correctly
```suggestion
In the absence of specification, a VQA block takes :class:`~.Qobj` a Hamiltonian :math:`H`, and will generate a unitary with free parameter, :math:`\theta`, as :math:`U(\theta) = e^{-i \theta H}`. For example, 
```

It would be nice if examples for other types of VQA blocks can also be added. Can leave it for another PR.